calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/query/DocumentDbMqlQueryContext.java

```diff
@@ -38,4 +40,18 @@
     private final String collectionName;
     /** The path information for the output documents. Maps column names to field paths.*/
     private final List<String> paths;
+
+    /**
+     * Gets the aggregation operations (stages) for the query as a list of strings.
+     *
+     * @return the aggregation operations as an ordered list of strings in extended JSON format.
+     */
+    public List<String> getAggregateOperationsAsStrings() {
+        return aggregateOperations.stream()
+                .map(
+                        doc ->
+                                doc.toBsonDocument()
+                                        .toJson(JsonWriterSettings.builder().outputMode(JsonMode.EXTENDED).build()))
```

Comment:
Using extended JSON format instead of toString() or default toJson() methods since these use standard JSON format. Extended JSON preserves BSON types.

Reply:
Nope. Just got carried away. 😅  Rearranged!